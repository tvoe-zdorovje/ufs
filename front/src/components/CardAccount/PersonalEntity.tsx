import React from 'react'
import {
  Row,
  Col,
  Button
} from 'antd'
import renderPair from './RenderPair'
import {
  AccountResidues,
  PersonalAccount,
  Seizure
} from 'interfaces/dto/Account.interface'

const styles = require('./CardAccount.less')

const getDetails = (accountResidues: AccountResidues) => ([['Текущий остаток', accountResidues.currentBalance],
    ['Неснижаемый остаток', accountResidues.fixedBalance],
    ['Планируемый остаток', accountResidues.expectedBalance]]
)

const getSeizures = (seizure) => (
  [['Сумма ареста', seizure.amount],
    ['Основание ареста ', seizure.reason],
    ['Срок ареста', `С ${seizure.fromDate} По ${seizure.toDate}`],
    ['ЮЛ/орган-инициатор', seizure.initiatorShortName],
  ])

const renderStatus = (info, accountResidues, addSidePanel, seizures) => (
  <Row className={(info.status === 'Active') ? styles.green : styles.red}>
    <Col span={8}>
      {(info.status === 'Active') ? 'Активный' : 'Неактивный'}
    </Col>
    <Col span={10} className={styles.rightAligned}>
      <Button
        className={`${styles.green} ${styles.noBorder}`}
        onClick={() => addSidePanel(getDetails(accountResidues), 'Доп. информация по счету')}
      >
        Подробнее
      </Button>
    </Col>
    <Col span={6}>
      {(seizures && seizures.length > 0) ? (
        <Button
          className={`${styles.red} ${styles.noBorder}`}
          onClick={() => addSidePanel(getSeizures(seizures[0]), 'Информация об арестах')}
        >
          Арест суммы
        </Button>) : null}
    </Col>
  </Row>
)

export interface PerosnalEntityProps {
  info: PersonalAccount,
  accountResidues: AccountResidues,
  addSidePanel: Function,
  seizures: Array<Seizure>,
}

export default function PerosnalEntity({ info, accountResidues, addSidePanel, seizures }: PerosnalEntityProps) {
  const pairs = [['Номер счета', info.id],
    ['Вид', 'Расчетный счет'],
    ['Номер договора', info.agreement.id],
    ['Текущий остаток', accountResidues.currentBalance],
    ['Валюта', info.currencyType],
    ['Дата последней операции', info.lastTransactionDate],
    ['Статус', renderStatus(info, accountResidues, addSidePanel, seizures)]]

  return (
    <Row>
      <Col span={23}>
        {pairs.map((item, i) => renderPair(item[0], item[1], i))}
      </Col>
    </Row>
  )
}
