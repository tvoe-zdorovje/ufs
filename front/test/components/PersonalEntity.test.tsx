import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

import PersonalEntity, { PerosnalEntityProps } from 'components/CardAccount/PersonalEntity'

describe('Компонент PersonalEntity', () => {

  const props: PerosnalEntityProps = {
    info: {
      id: 1,
      typeId: 1,
      currencyType: 'RUB',
      agreement: {
        id: '1',
        openDate: '1',
      },
      tbCode: '1',
      gosbCode: '1',
      vspCode: '1',
      subbranchCode: '1',
      lastTransactionDate: '1',
      active: true,
      status: '1',
      changeStatusDescription: '1',
      clientTypeFk: true,
      seizured: true
    },
    accountResidues: {
      currentBalance: '1',
      expectedBalance: '1',
      fixedBalance: '1',
    },
    addSidePanel: (x, y) => {
    },
    seizures: [{
      type: 1,
      reason: '1',
      fromDate: '1',
      toDate: '1',
      amount: '1',
      initiatorShortName: '1',
    }]
  }

  const renderWrapper = render(<PersonalEntity {...props} />)

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
