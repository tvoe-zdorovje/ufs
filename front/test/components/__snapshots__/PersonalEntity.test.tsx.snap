// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент PersonalEntity совпадает со своим снимком 1`] = `
<div
  class="ant-row"
>
  <div
    class="ant-col-23"
  >
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Номер счета
      </div>
      <div
        class="ant-col-16"
      >
        1
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Вид
      </div>
      <div
        class="ant-col-16"
      >
        Расчетный счет
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Номер договора
      </div>
      <div
        class="ant-col-16"
      >
        1
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Текущий остаток
      </div>
      <div
        class="ant-col-16"
      >
        1
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Валюта
      </div>
      <div
        class="ant-col-16"
      >
        RUB
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Дата последней операции
      </div>
      <div
        class="ant-col-16"
      >
        1
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Статус
      </div>
      <div
        class="ant-col-16"
      >
        <div
          class="ant-row red"
        >
          <div
            class="ant-col-8"
          >
            Неактивный
          </div>
          <div
            class="ant-col-10 rightAligned"
          >
            <button
              class="ant-btn green noBorder"
              type="button"
            >
              <span>
                Подробнее
              </span>
            </button>
          </div>
          <div
            class="ant-col-6"
          >
            <button
              class="ant-btn red noBorder"
              type="button"
            >
              <span>
                Арест суммы
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
